etc/Makefile.am

```diff
@@ -40,5 +42,20 @@ pkgconfig_DATA = flux-core.pc \
 	flux-schedutil.pc
 endif
 
+noinst_SCRIPTS = \
+	completions/get_builtins.sh \
+	completions/flux.pre
+
 EXTRA_DIST = \
-	gen-cmdhelp.py
+	gen-cmdhelp.py \
+	$(noinst_SCRIPTS)
+
+completions/flux: $(srcdir)/completions/flux.pre
+	$(AM_V_GEN)test -d completions || mkdir completions && \
+	cp $< $@ && chmod +w $@ && \
+	$(srcdir)/completions/get_builtins.sh \
+	$(top_srcdir)/src/cmd/builtin >> $@
+
+fluxcompdir = $(sysconfdir)/bash_completion.d
```

Comment:
I noticed on my Ubuntu system that there is a `bash-completion.pc` pkg-config file, and that it shows:
```
$ cat /usr/share/pkgconfig/bash-completion.pc 
prefix=/usr
compatdir=/etc/bash_completion.d
completionsdir=${prefix}/share/bash-completion/completions
helpersdir=${prefix}/share/bash-completion/helpers

Name: bash-completion
Description: programmable completion for the bash shell
URL: https://github.com/scop/bash-completion
Version: 2.8
```

Seems like `fluxcompdir` should be set via pkg-config lookup for `bash-completion`, falling back to `$(sysconfdir)/bash_completion.d` if not found.

Reply:
Oh good find!

> Seems like `fluxcompdir` should be set via pkg-config lookup for `bash-completion`, falling back to `$(sysconfdir)/bash_completion.d` if not found.

Hmm, but that wouldn't be right for a side-install, e.g. `make install` would fail with permission denied for, say, `--prefix=$HOME/local`.

Maybe we should go ahead and merge this, but open an issue to discuss the right way to use `bash-completion.pc`.  Off the top of my head I'm not having any great ideas...

Reply:
> Hmm, but that wouldn't be right for a side-install, e.g. make install would fail with permission denied for, say, --prefix=$HOME/local.

Ah, I hadn't considered that. My main discovery was that `$(sysconfdir)/bash_completion.d` is the "compat" directory, and the real bash completion path is `${prefix}/share/bash-completion/completions`

Reply:
Hm, apparently you can do this with `pkg-config`:
```
$ pkg-config --define-variable=prefix=/usr/local --variable=completionsdir bash-completion
/usr/local/share/bash-completion/completions
```